api/service.go

```diff
@@ -0,0 +1,391 @@
+// treasury.
+func (s *Service) GetTreasuryDetails() (treasuryDetails *TreasuryDetails, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/treasury/balance",
+	}
+	tDetails := &TreasuryDetails{}
+	return tDetails, s.client.Do(DcrData, s.chainParams.Name, reqConf, tDetails)
+}
+
+// GetExchangeRate fetches exchange rate data summary
+func (s *Service) GetExchangeRate() (rates *ExchangeRates, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/exchangerate",
+	}
+	exRates := &ExchangeRates{}
+	// Use mainnet base url for exchange rate endpoint
+	return exRates, s.client.Do(DcrData, chaincfg.MainNetParams().Name, reqConf, exRates)
+}
+
+// GetExchanges fetches the current known state of all exchanges
+func (s *Service) GetExchanges() (state *ExchangeState, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/exchanges",
+	}
+	exState := &ExchangeState{}
+	// Use mainnet base url for exchanges endpoint
+	return exState, s.client.Do(DcrData, chaincfg.MainNetParams().Name, reqConf, exState)
+}
+
+// GetTicketFeeRateSummary returns the current ticket fee rate summary. See dcrdata's MempoolTicketFeeInfo for the specific
+// data returned.
+func (s *Service) GetTicketFeeRateSummary() (ticketInfo *apiTypes.MempoolTicketFeeInfo, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/mempool/sstx",
+	}
+	tFeeSum := &apiTypes.MempoolTicketFeeInfo{}
+	return tFeeSum, s.client.Do(DcrData, s.chainParams.Name, reqConf, tFeeSum)
+}
+
+// GetTicketFeeRate returns top 25 ticket fees. Note: in cases where n < 25 and n == number of all ticket fees,
+// It returns n.
+func (s *Service) GetTicketFeeRate() (ticketFeeRate *apiTypes.MempoolTicketFees, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/mempool/sstx/fees",
+	}
+	tFeeRate := &apiTypes.MempoolTicketFees{}
+	return tFeeRate, s.client.Do(DcrData, s.chainParams.Name, reqConf, tFeeRate)
+}
+
+// GetNHighestTicketFeeRate returns the {nHighest} ticket fees. For cases where total number of ticker is less than
+// {nHighest} it returns the fee rate for the total number of tickets.
+func (s *Service) GetNHighestTicketFeeRate(nHighest int) (ticketFeeRate *apiTypes.MempoolTicketFees, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/mempool/sstx/fees/" + strconv.Itoa(nHighest),
+	}
+	nTFeeRate := &apiTypes.MempoolTicketFees{}
+	return nTFeeRate, s.client.Do(DcrData, s.chainParams.Name, reqConf, nTFeeRate)
+}
+
+// GetTicketDetails returns all ticket details see drcdata's MempoolTicketDetails for the spcific information
+// returned.
+func (s *Service) GetTicketDetails() (ticketDetails *apiTypes.MempoolTicketDetails, err error) {
+	reqConf := &ReqConfig{
+		method: http.MethodGet,
+		url:    "api/mempool/sstx/details",
+	}
+	tDetails := &apiTypes.MempoolTicketDetails{}
```

Comment:
i don't think there's any advantage in shortening this. `ticketDetails` is a better variable name, and it's not an unusually long name that we need to shorten it, i saw similar instances of this such as `aList` instead of `agendaList` etc. not a worthy trade off in my opinion.

It's not a blocker in getting this merged, just thought to mention

Reply:
The shortened variable are temporary variables used only in the functions. You can see the longer versions in the method signature.

Reply:
Good catch. I'll fix that.